client/src/components/Auth/AuthProvider.js

```diff
@@ -0,0 +1,86 @@
+import React, { useState } from 'react'
+import axios from 'axios'
+import jwt from 'jwt-decode'
+
+const AuthContext = React.createContext(null)
+
+const AuthProvider = ({ children }) => {
+    const [auth, setAuth] = useState(null)
+
+    const login = async (email, password, callback) => { 
+        console.log("[Login]")
+        try{
+            const authResponse = await axios.post(
+                'http://localhost:8000/auth/login', 
+                { email: email, password: password }, 
+                { 'content-type': 'application/json' }
+            )
+            const decoded = jwt(authResponse.data.token)
+            setAuth({ token: authResponse.data.token, user: decoded.user })
+            console.log(auth.token)
+            callback()
+        } catch (err) {
+            console.log(`Login error ${err}`)
+            callback(err)
+        }
+    }
+
+    const register = async (firstName, lastName, email, password, callback) => { 
+        // Assignment: how do we register someone?
+        console.log("[Register]")
+        try{
+            const authResponse = await axios.post(
+                'http://localhost:8000/auth/register',
+                {   
+                    firstName: firstName,
+                    lastName: lastName,
+                    email: email,
+                    password: password
+                },
+                { 'content-type': 'application/json' }
+            )
+            const decoded = jwt(authResponse.data.token)
+            setAuth({ token: authResponse.data.token, user: decoded.user})
+            axios.defaults.headers.common['Authorization'] = 'Bearer ' + auth.token
+            console.log('Bearer ' + auth.token)
+            try{
+                const res = await axios.post(
```

Comment:
I like that you're trying to create a demo deck on registration. I would suggest that it is easier to have a user redirected to login instead of trying to get an auth token on registration and then route to a create deck component immediately.